### ISSUE TITLE
3.2.3 crashes on Mojave 10.14.2

### ISSUE THREAD

**Issue by arisc097** [HappenApps/Quiver#1248]
3.1.3 works fine. The issue is with the latest version . Attempt to delete a note crashes the app consistently. 

Process:               Quiver [549]
Path:                  /Applications/Quiver.app/Contents/MacOS/Quiver
Identifier:            com.happenapps.Quiver
Version:               3.2.3 (188)
App Item ID:           866773894
App External ID:       829552550
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Quiver [549]
User ID:               501

Date/Time:             2018-12-22 21:30:39.489 -0700
OS Version:            Mac OS X 10.14.2 (18C54)
Report Version:        12
Anonymous UUID:        04373B0A-5705-0D10-FD0A-5BC699EA5451

**Comment by ylian** [comment]
Could you post the entire crash log? I haven't yet received any other report like this.

**Comment by arisc097** [comment]
I am unable to reproduce it.  The latest crash:

``` Process:               Quiver [1453]
Path:                  /Applications/Quiver.app/Contents/MacOS/Quiver
Identifier:            com.happenapps.Quiver
Responsible:           Quiver [1453]
    __TEXT                 00000001002b1000-00000001003a1000 [  960K] r-x/rwx SM=COW  /Applications/Quiver.app/Contents/MacOS/Quiver
0   com.happenapps.Quiver         	0x000000010033d42a 0x1002b1000 + 574506
1   com.happenapps.Quiver         	0x000000010033d798 0x1002b1000 + 575384
2   com.happenapps.Quiver         	0x00000001002de279 0x1002b1000 + 184953
9   com.happenapps.Quiver         	0x00000001003147a6 0x1002b1000 + 407462
       0x1002b1000 -        0x1003a0fff +com.happenapps.Quiver (3.2.3 - 188) <127632E1-898D-3D18-B30F-6FA4EDA952A2> /Applications/Quiver.app/Contents/MacOS/Quiver
       0x1003ff000 -        0x1005b4fff +com.github.groue.GRDB (3.5.0 - 1) <E22FA6B9-88E0-3247-8C58-9F8EA8E93163> /Applications/Quiver.app/Contents/Frameworks/GRDB.framework/Versions/A/GRDB
       0x100721000 -        0x100735ff7 +libswiftAppKit.dylib (4.2.1 - 1000.11.42) <1223BA92-5392-33F4-893F-8C20A5D0905D> /Applications/Quiver.app/Contents/Frameworks/libswiftAppKit.dylib
       0x10075f000 -        0x100af1ff7 +libswiftCore.dylib (4.2.1 - 1000.11.42) <D0C86227-BFC8-30D7-93B8-9F228E0F15BA> /Applications/Quiver.app/Contents/Frameworks/libswiftCore.dylib
       0x100dc9000 -        0x100dd1fff +libswiftCoreData.dylib (4.2.1 - 1000.11.42) <6CC602C0-1AA9-3DD3-B843-CD1D07C36F27> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreData.dylib
       0x100de3000 -        0x100de6fff +libswiftCoreFoundation.dylib (4.2.1 - 1000.11.42) <801E1416-6C23-3C48-BF46-2349B9F36C9E> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreFoundation.dylib
       0x100df2000 -        0x100e01ff7 +libswiftCoreGraphics.dylib (4.2.1 - 1000.11.42) <579D172D-0873-33EB-A529-742D5D70BD74> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreGraphics.dylib
       0x100e2a000 -        0x100e2fff7 +libswiftCoreImage.dylib (4.2.1 - 1000.11.42) <4DB8ED84-0EBB-3922-B3A4-82EE4834549B> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreImage.dylib
       0x100e39000 -        0x100e41fff +libswiftDarwin.dylib (4.2.1 - 1000.11.42) <70D5D45B-0327-352C-BDCE-F792699DAB88> /Applications/Quiver.app/Contents/Frameworks/libswiftDarwin.dylib
       0x100e57000 -        0x100e70ff7 +libswiftDispatch.dylib (4.2.1 - 1000.11.42) <E8DBE923-0513-3FE8-8E3A-5778014FD810> /Applications/Quiver.app/Contents/Frameworks/libswiftDispatch.dylib
       0x100eb5000 -        0x10101dfff +libswiftFoundation.dylib (4.2.1 - 1000.11.42) <2252DB9E-6B26-31AD-A869-9D92FA2BE196> /Applications/Quiver.app/Contents/Frameworks/libswiftFoundation.dylib
       0x1011b9000 -        0x1011bcfff +libswiftIOKit.dylib (4.2.1 - 1000.11.42) <575A9454-3EB8-3314-B67B-9D0C88C03D5F> /Applications/Quiver.app/Contents/Frameworks/libswiftIOKit.dylib
       0x1011c9000 -        0x1011d2fff +libswiftMetal.dylib (4.2.1 - 1000.11.42) <A538B219-717C-30F5-BB27-BF1C170EE99F> /Applications/Quiver.app/Contents/Frameworks/libswiftMetal.dylib
       0x1011e3000 -        0x1011e7fff +libswiftObjectiveC.dylib (4.2.1 - 1000.11.42) <A521AAA2-3B1F-38AF-BFA0-BB02A94ECA8B> /Applications/Quiver.app/Contents/Frameworks/libswiftObjectiveC.dylib
       0x1011f9000 -        0x1011ffff7 +libswiftQuartzCore.dylib (4.2.1 - 1000.11.42) <C619FC2F-405B-3E38-8FDA-36B2A9EBA0F2> /Applications/Quiver.app/Contents/Frameworks/libswiftQuartzCore.dylib
       0x10120d000 -        0x101210fff +libswiftXPC.dylib (4.2.1 - 1000.11.42) <3EC739CE-A1AC-3199-8461-9CA872E2B9EF> /Applications/Quiver.app/Contents/Frameworks/libswiftXPC.dylib
       0x10121f000 -        0x101226fff +libswiftos.dylib (4.2.1 - 1000.11.42) <0B29ACF1-EFFD-3F02-B509-5210DC92DD31> /Applications/Quiver.app/Contents/Frameworks/libswiftos.dylib
```

The earliest crash right after upgrade:
``` Process:               Quiver [651]
Path:                  /Applications/Quiver.app/Contents/MacOS/Quiver
Identifier:            com.happenapps.Quiver
Responsible:           Quiver [651]
    __TEXT                 000000010bd5d000-000000010be4d000 [  960K] r-x/rwx SM=COW  /Applications/Quiver.app/Contents/MacOS/Quiver
0   com.happenapps.Quiver         	0x000000010bde942a 0x10bd5d000 + 574506
1   com.happenapps.Quiver         	0x000000010bde9798 0x10bd5d000 + 575384
2   com.happenapps.Quiver         	0x000000010bd8a279 0x10bd5d000 + 184953
3   com.happenapps.Quiver         	0x000000010bd8de6c 0x10bd5d000 + 200300
30  com.happenapps.Quiver         	0x000000010bdc07a6 0x10bd5d000 + 407462
       0x10bd5d000 -        0x10be4cfff +com.happenapps.Quiver (3.2.3 - 188) <127632E1-898D-3D18-B30F-6FA4EDA952A2> /Applications/Quiver.app/Contents/MacOS/Quiver
       0x10beab000 -        0x10c060fff +com.github.groue.GRDB (3.5.0 - 1) <E22FA6B9-88E0-3247-8C58-9F8EA8E93163> /Applications/Quiver.app/Contents/Frameworks/GRDB.framework/Versions/A/GRDB
       0x10c1cd000 -        0x10c1e1ff7 +libswiftAppKit.dylib (4.2.1 - 1000.11.42) <1223BA92-5392-33F4-893F-8C20A5D0905D> /Applications/Quiver.app/Contents/Frameworks/libswiftAppKit.dylib
       0x10c20c000 -        0x10c59eff7 +libswiftCore.dylib (4.2.1 - 1000.11.42) <D0C86227-BFC8-30D7-93B8-9F228E0F15BA> /Applications/Quiver.app/Contents/Frameworks/libswiftCore.dylib
       0x10c878000 -        0x10c880fff +libswiftCoreData.dylib (4.2.1 - 1000.11.42) <6CC602C0-1AA9-3DD3-B843-CD1D07C36F27> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreData.dylib
       0x10c893000 -        0x10c896fff +libswiftCoreFoundation.dylib (4.2.1 - 1000.11.42) <801E1416-6C23-3C48-BF46-2349B9F36C9E> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreFoundation.dylib
       0x10c89e000 -        0x10c8adff7 +libswiftCoreGraphics.dylib (4.2.1 - 1000.11.42) <579D172D-0873-33EB-A529-742D5D70BD74> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreGraphics.dylib
       0x10c8d3000 -        0x10c8d8ff7 +libswiftCoreImage.dylib (4.2.1 - 1000.11.42) <4DB8ED84-0EBB-3922-B3A4-82EE4834549B> /Applications/Quiver.app/Contents/Frameworks/libswiftCoreImage.dylib
       0x10c8e2000 -        0x10c8eafff +libswiftDarwin.dylib (4.2.1 - 1000.11.42) <70D5D45B-0327-352C-BDCE-F792699DAB88> /Applications/Quiver.app/Contents/Frameworks/libswiftDarwin.dylib
       0x10c904000 -        0x10c91dff7 +libswiftDispatch.dylib (4.2.1 - 1000.11.42) <E8DBE923-0513-3FE8-8E3A-5778014FD810> /Applications/Quiver.app/Contents/Frameworks/libswiftDispatch.dylib
       0x10c95c000 -        0x10cac4fff +libswiftFoundation.dylib (4.2.1 - 1000.11.42) <2252DB9E-6B26-31AD-A869-9D92FA2BE196> /Applications/Quiver.app/Contents/Frameworks/libswiftFoundation.dylib
       0x10cc60000 -        0x10cc63fff +libswiftIOKit.dylib (4.2.1 - 1000.11.42) <575A9454-3EB8-3314-B67B-9D0C88C03D5F> /Applications/Quiver.app/Contents/Frameworks/libswiftIOKit.dylib
       0x10cc6f000 -        0x10cc78fff +libswiftMetal.dylib (4.2.1 - 1000.11.42) <A538B219-717C-30F5-BB27-BF1C170EE99F> /Applications/Quiver.app/Contents/Frameworks/libswiftMetal.dylib
       0x10cc87000 -        0x10cc8bfff +libswiftObjectiveC.dylib (4.2.1 - 1000.11.42) <A521AAA2-3B1F-38AF-BFA0-BB02A94ECA8B> /Applications/Quiver.app/Contents/Frameworks/libswiftObjectiveC.dylib
       0x10cc99000 -        0x10cc9fff7 +libswiftQuartzCore.dylib (4.2.1 - 1000.11.42) <C619FC2F-405B-3E38-8FDA-36B2A9EBA0F2> /Applications/Quiver.app/Contents/Frameworks/libswiftQuartzCore.dylib
       0x10ccad000 -        0x10ccb0fff +libswiftXPC.dylib (4.2.1 - 1000.11.42) <3EC739CE-A1AC-3199-8461-9CA872E2B9EF> /Applications/Quiver.app/Contents/Frameworks/libswiftXPC.dylib
       0x10ccbe000 -        0x10ccc5fff +libswiftos.dylib (4.2.1 - 1000.11.42) <0B29ACF1-EFFD-3F02-B509-5210DC92DD31> /Applications/Quiver.app/Contents/Frameworks/libswiftos.dylib
```

**Comment by arisc097** [comment]
- Also in the previous version, the current library was mapped to iCloud Drive. 
- When I upgrade to 3.2.3, the current library is mapped to ``` /Users/<user>/Library/Containers/com.happenapps.Quiver/Data/Library/Application Support/Quiver/Quiver.qvlibrary```.

The upgrade doesn't automatically map to existing current library. When I try to remap to old library by doing ```Quiver - Preference - Sync - Click on right arrow on Current Library```, the experience is bit odd. I double-click on the old library, and then it opens another instance of Quiver.  When I close both instances, the remap choice isn't remembered. This cause me to move the current library content between two Quiver active instances. 

https://drive.google.com/open?id=13hT836tOmIDyOGL0tBzDgdyetB3j-B0Y

**Comment by ylian** [comment]
@arisc097 I see. Somehow the upgrade didn't map to the previously selected library. The new version supports opening multiple libraries at the same time, so the workflow is a little different. Simply double click on a Quiver library to open it, or use File Menu > Open Library. The right arrow in the Sync tab is to show the currently selected library, not for changing library location.

Quiver remembers window configurations, so relaunching Quiver will open both windows. If you just want one window, simply close the other window.

**Comment by arisc097** [comment]
@ylian Thanks.

**Comment by ylian** [comment]
Fixed the crash in 3.2.5.
